src/euphorie/client/browser/consultancy.py

```diff
@@ -256,7 +256,12 @@ def handle_POST(self):
             )
             self.sqlsession.add(event)
             self.context.session.consultancy.status = "validated"
-            # TODO: lock session
+            locking_view = api.content.get_view(
+                name="locking_view",
+                context=self.context,
+                request=self.request,
+            )
+            locking_view.create_event("lock_set")
```

Comment:
I do not think we need to create a lock event, we already have the validated event

Reply:
The validated event is not in the list of locking events, so it is ignored by the locking view.

Reply:
You probably missed https://github.com/euphorie/Euphorie/pull/577

Reply:
Actually I think we want a condition here. If the assessment has already been locked by an admin before it is validated, then we don't want to lock it again here. We want to show the time and user name from the original locking event in the locking menu.

Reply:
> You probably missed #577

I am aware of it but it does not fulfill the requirements.

Reply:
Can you please update your PR to cope with the fact that a lock can be there either because there was a "lock_set" action and a "validated" action?

Reply:
I've updated the code. Please see the tests for why we need it this way and ping me if you need further explanation.